File: src/main/java/com/kingdom/util/CardCostComparator.java
package com.kingdom.util;

import com.kingdom.model.Card;

import java.util.Comparator;

/**
 * Created by IntelliJ IDEA.
 * User: John
 * Date: Aug 18, 2010
 * Time: 7:10:29 AM
 */
public class CardCostComparator implements Comparator<Card> {
    public int compare(Card c1, Card c2) {
        int cost1 = c1.getCost();
        if (c1.isCostIncludesPotion()) {
            cost1 += 2;
        }
        int cost2 = c2.getCost();
        if (c2.isCostIncludesPotion()) {
            cost2 += 2;
        }

        if (cost1 > cost2) {
            return +1;
        }
        else if (cost1 < cost2) {
            return -1;
        }
        else {
            return 0;
        }
    }
}

File: src/main/kotlin/com/kingdom/util/CardCostComparator.kt
package com.kingdom.util

import com.kingdom.model.cards.Card

import java.util.Comparator

class CardCostComparator : Comparator<Card> {
    override fun compare(c1: Card, c2: Card): Int {
        val cost1 = c1.cost
        val cost2 = c2.cost

        return when {
            cost1 > cost2 -> 1
            cost1 < cost2 -> -1
            else -> 0
        }
    }
}
